site-src/geps/gep-1323.md

```diff
@@ -0,0 +1,274 @@
+      backendRefs:
+      - name: example-svc
+        weight: 1
+        port: 80
+```
+
+## Prior Art
+A few projects that implement Gateway API already have support for similar functionality (in their custom CRDs), like:
+* Istio’s `VirtualService`:
+
+```yaml
+apiVersion: networking.istio.io/v1alpha3
+kind: VirtualService
+metadata:
+  name: reviews-route
+spec:
+  hosts:
+  - reviews.prod.svc.cluster.local
+  http:
+  - headers:
+      request:
+        set:
+          test: "true"
+    route:
+    - destination:
+        host: reviews.prod.svc.cluster.local
+        subset: v2
+      weight: 25
+    - destination:
+        host: reviews.prod.svc.cluster.local
+        subset: v1
+      headers:
+        response:
+          remove:
+          - foo
+      weight: 75
+```
+
+* Contour’s `HTTPProxy`:
+
+```yaml
+apiVersion: projectcontour.io/v1
+kind: HTTPProxy
+metadata:
+  name: basic
+spec:
+  virtualhost:
+    fqdn: foo-basic.bar.com
+  routes:
+    - conditions:
+      - prefix: /
+      services:
+        - name: s1
+          port: 80
+      responseHeadersPolicy:
+        set:
+          name: test
+          value: true
+```
+
+* Ingress NGINX:
+
+```yaml
+apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata:
+  name: nginx-headers
+  annotations:
+    nginx.ingress.kubernetes.io/configuration-snippet: |
+      add_header ingress nginx;
+spec:
+  ingressClassName: nginx
+  rules:
+  - host: custom.configuration.com
+    http:
+      paths:
+      - path: /
+        pathType: Prefix
+        backend:
+          service:
+            name: http-svc
+            port:
+              number: 80
+```
+
+## Policy Attachment
+
+This feature can be further extended via [Policy Attachment](../references/policy-attachment.md). Implementations can introduce a new Policy, `ResponseHeaderPolicy`, which can be used to ensure that certain headers are present/absent in a response, regardless of the `responseHeaderModifier` field in the `HTTPRoute` associated with that response.
+
+```golang
+type ResponseHeaderPolicy struct {
+    metav1.TypeMeta   `json:",inline"`
+    metav1.ObjectMeta `json:"metadata,omitempty"`
+
+    // Spec defines the desired state of ResponseHeaderPolicy.
+    Spec ResponseHeaderPolicySpec `json:"spec"`
+
+    // Status defines the current state of ResponseHeaderPolicy.
+    Status ResponseHeaderPolicyStatus `json:"status,omitempty"`
+}
+
+type ResponseHeaderPolicySpec struct {
+    // TargetRef identifies an API object to apply policy to.
+    TargetRef gatewayv1a2.PolicyTargetReference `json:"targetRef"`
+
+    // Override defines policy configuration that should override policy
+    // configuration attached below the targeted resource in the hierarchy.
+    // +optional
+    Override *HTTPHeaderModifier `json:"override,omitempty"`
+
+    // Default defines default policy configuration for the targeted resource.
+    // +optional
+    Default *HTTPHeaderModifier `json:"default,omitempty"`
```

Comment:
I don't think `Default` is needed here.  If so, how is the `Default` different than the `Override`?  What happens if both are specified?

Reply:
I think the example below describes the usage of `Default` and `Override` sufficiently. About both being specified, I don't think that makes any sense, so maybe I should add something like _only_ one of `Default` and `Override` should be present?

Reply:
https://gateway-api.sigs.k8s.io/references/policy-attachment/#policy-boilerplate does describe allowing both `Default` and `Override` to be set, although I think that often makes more sense when they modify different values, such as the `AcmeServicePolicy` example in https://gateway-api.sigs.k8s.io/references/policy-attachment/#hierarchy